homeassistant/components/wled/config_flow.py

```diff
@@ -72,23 +77,22 @@ class WLEDFlowHandler(ConfigFlow, domain=DOMAIN):
         if source == SOURCE_ZEROCONF:
             # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
             user_input[CONF_HOST] = self.context.get(CONF_HOST)
-
-        errors = {}
-        session = async_get_clientsession(self.hass)
-        wled = WLED(user_input[CONF_HOST], session=session)
-
-        try:
-            device = await wled.update()
-        except WLEDConnectionError:
-            if source == SOURCE_ZEROCONF:
-                return self.async_abort(reason="connection_error")
-            errors["base"] = "connection_error"
-            return self._show_setup_form(errors)
+            user_input[CONF_MAC] = self.context.get(CONF_MAC)
+
+        if user_input.get(CONF_MAC) is None or not prepare:
+            session = async_get_clientsession(self.hass)
+            wled = WLED(user_input[CONF_HOST], session=session)
+            try:
+                device = await wled.update()
+            except WLEDConnectionError:
+                if source == SOURCE_ZEROCONF:
+                    return self.async_abort(reason="connection_error")
+                return self._show_setup_form({"base": "connection_error"})
+            user_input[CONF_MAC] = device.info.mac_address
```

Comment:
It was a bit confusing that data coming from the device is being stored in `user_input`.  Its not a case where untrusted user data is making into data we trust so not likely to be an issue

Reply:
That is not a choice I made for this integration, it just following the base classes and developer documentation on this.

From a perspective, one could argue the naming, since it is not data the input using a keyboard, however, it is provided/put in by the user's network... 🤷‍♂